core/src/queue.rs

```diff
@@ -80,12 +80,12 @@ impl<W: WorldTrait> Queue<W> {
     }
 
     /// Returns `n` randomly selected transaction from the queue.
-    pub fn n_random_transactions(&self, n: usize) -> Vec<VersionedAcceptedTransaction> {
+    pub fn n_random_transactions(&self, n: u32) -> Vec<VersionedAcceptedTransaction> {
         self.txs
             .iter()
             .filter(|e| self.is_pending(e.value()))
             .map(|e| e.value().clone())
-            .choose_multiple(&mut rand::thread_rng(), n)
+            .choose_multiple(&mut rand::thread_rng(), n as usize)
     }
```

Comment:
I think it should be this method user responsibility to provide correct `usize` number. So I think this method should be left unchanged

Reply:
I don't think we want to write code that supports certain behavior on certain platforms. Providing a value bigger than U32_MAX should not be allowed because that won't run on 32bit, better make the caller aware of this through a 32bit argument.